src/geom/SpanSet.cc

```diff
@@ -692,9 +692,13 @@ std::shared_ptr<SpanSet> SpanSet::fromShape(int r, Stencil s, lsst::geom::Point2
     tempVec.reserve(2 * r + 1);
     switch (s) {
         case Stencil::CIRCLE:
-            for (auto dy = -r; dy <= r; ++dy) {
-                int dx = static_cast<int>(sqrt(r * r - dy * dy));
-                tempVec.push_back(Span(dy + offset.getY(), -dx + offset.getX(), dx + offset.getX()));
+            {
+                double dr = static_cast<double>(r);
```

Comment:
I think a comment is necessary to clarify what problem is being avoided by the parallel increment of the integer and the double.  And I might name the variable that holds "r-squared" something like `r_sq`.